### PR TITLE
[QA-1992] test: edge app testing migration to azion-console-kit

### PR DESCRIPTION
**Pull request by andradepradorafael**
### Does this PR introduce breaking changes?
- [X] No
- [ ] Yes 

### Does this PR introduce UI changes? Add a video or screenshots here.
No

### Does it have a link on Figma?
No

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [X] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [X] Commits are tagged with the right word (feat, test, refactor, etc)
- [X] Application responsiveness was tested to different screen sizes
- [X] Code is formatted and linted
- [X] Tags are added to the PR

#### These changes were tested on the following browsers:
- [X] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Review by william-tome — CHANGES_REQUESTED** [review]
O linter da sua máquina está sobrescrevendo as regras de lint do projeto.

**Review by william-tome — APPROVED** [review]

File: cypress.config.js [hunk 1]
```diff
@@ -7,13 +7,9 @@ export default defineConfig({
...
+    defaultCommandTimeout: 15000,
     video: true,
     experimentalStudio: true,
-    retries: {
-      runMode: 2,
-      openMode: 2
-    },
     setupNodeEvents(on, config) {
       codeCoverageTask(on, config);
       return config;
```
Comment (william-tome) [inline cypress.config.js:15]:
```suggestion
      codeCoverageTask(on, config)
      return config
```

File: cypress/e2e/edge_application.cy.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    cy.visit('/edge-applications')
+    cy.get(selectors.edgeApplication.searchInput).type(edgeApplicationName)
+    cy.get(selectors.edgeApplication.tableRowName).should('be.visible').should('have.text', edgeApplicationName)
+    cy.get(selectors.edgeApplication.actionsButton).click()
+    cy.get(selectors.edgeApplication.deleteButton).click()
+    cy.get(selectors.edgeApplication.confirmDelete).click()
+    cy.get(selectors.edgeApplication.deleteInput).type('delete')
+    cy.get(selectors.edgeApplication.confirmDeleteButton).click()
+  })
+})
```
Comment (william-tome) [inline cypress/e2e/edge_application.cy.js:104]:
```suggestion
  const timestamp = Date.now()
  return `${prefix}-${timestamp}`
};

const email = Cypress.env('CYPRESS_EMAIL_STAGE')
const password = Cypress.env('CYPRESS_PASSWORD_STAGE')
const edgeApplicationName = generateUniqueName('EdgeApp')
const rulesEngineName = generateUniqueName('RulesEng')

describe('Edge Application', () => {
  beforeEach(() => {
    // Login
    cy.visit('/login')
    cy.get(selectors.login.emailInput).type(email)
    cy.get(selectors.login.nextButton).click()
    cy.get(selectors.login.passwordInput).type(password)
    cy.get(selectors.login.signInButton).click()
  })

  it('Create and delete an edge application, and create a rule', () => {
    // Navigate to Edge Applications
    cy.get(selectors.sidebar.menuToggleButton).click()
    cy.get(selectors.sidebar.edgeApplicationMenuItem).click()

    // Create an edge application
    cy.get(selectors.edgeApplication.createButton).click()
    cy.get(selectors.edgeApplication.nameInput).type(edgeApplicationName)
    cy.get(selectors.edgeApplication.addressInput).clear()
    cy.get(selectors.edgeApplication.addressInput).type('httpbingo.org')
    cy.get(selectors.edgeApplication.saveButton).click()
    cy.get(selectors.edgeApplication.cancelButton).click()

    // Verify the edge application was created
    cy.get(selectors.edgeApplication.searchInput).type(edgeApplicationName)
    cy.get(selectors.edgeApplication.tableRowName).should('be.visible').should('have.text', edgeApplicationName)

    // Navigate to Rules Engine Tab
    cy.get(selectors.edgeApplication.tableRowLastEditor).click()
    cy.get(selectors.edgeApplication.rulesEngineTab).click()

    // Create a rule
    cy.get(selectors.edgeApplication.addRuleButton).click()
    cy.get(selectors.edgeApplication.ruleNameInput).type(rulesEngineName)
    cy.get(selectors.edgeApplication.criteriaOperatorDropdown).click()
    cy.get(selectors.edgeApplication.criteriaOperator).click()
    cy.get(selectors.edgeApplication.criteriaInputValue).clear('/')
    cy.get(selectors.edgeApplication.criteriaInputValue).type('/')
    cy.get(selectors.edgeApplication.behaviorsDropdown).click()
    cy.get(selectors.edgeApplication.behaviors).click()
    cy.get(selectors.edgeApplication.saveButton).click()
    cy.get('.flex-column > .text-sm').should('be.visible')

    // Verify the rule was created
    cy.get(selectors.edgeApplication.searchInput).type(rulesEngineName)
    cy.get(selectors.edgeApplication.ruleTable).should('be.visible').should('have.text', rulesEngineName)

    // Delete the edge application
    cy.visit('/edge-applications')
    cy.get(selectors.edgeApplication.searchInput).type(edgeApplicationName)
    cy.get(selectors.edgeApplication.tableRowName).should('be.visible').should('have.text', edgeApplicationName)
    cy.get(selectors.edgeApplication.actionsButton).click()
    cy.get(selectors.edgeApplication.deleteButton).click()
    cy.get(selectors.edgeApplication.confirmDelete).click()
    cy.get(selectors.edgeApplication.deleteInput).type('delete')
    cy.get(selectors.edgeApplication.confirmDeleteButton).click()
  })
})
```

File: cypress/e2e/variables.cy.js [hunk 3]
```diff
@@ -1,36 +1,75 @@
...
+    // Login
+    cy.visit('/login')
+    cy.get(selectors.login.emailInput).type(email)
+    cy.get(selectors.login.nextButton).click()
+    cy.get(selectors.login.passwordInput).type(password)
+    cy.get(selectors.login.signInButton).click()
+
+    // Navegar até a tela de Variáveis
+    cy.get(selectors.sidebar.menuToggleButton).click();
+    cy.get(selectors.sidebar.variablesMenuItem).click();
```
Comment (william-tome) [inline cypress/e2e/variables.cy.js:48]:
```suggestion
    cy.get(selectors.sidebar.menuToggleButton).click()
    cy.get(selectors.sidebar.variablesMenuItem).click()
```

File: cypress/e2e/variables.cy.js [hunk 4]
```diff
@@ -1,36 +1,75 @@
...
-    cy.get('#confirm-input').clear();
-    cy.get('#confirm-input').type('delete');
-    cy.get('.p-button-danger > .p-button-label').click();
-    cy.get('.justify-between > .flex > .text-color').should('have.text', 'Variable successfully deleted');
+    cy.get(selectors.variables.actionsButton).click();
+    cy.get(selectors.variables.deleteButton).click();
+    cy.get(selectors.variables.deleteInput).clear();
+    cy.get(selectors.variables.deleteInput).type('delete');
+    cy.get(selectors.variables.confirmDeleteButton).click();
+    cy.get(selectors.variables.successMessage).should('have.text', 'Variable successfully deleted');
```
Comment (william-tome) [inline cypress/e2e/variables.cy.js:73]:
```suggestion
    cy.get(selectors.variables.actionsButton).click()
    cy.get(selectors.variables.deleteButton).click()
    cy.get(selectors.variables.deleteInput).clear()
    cy.get(selectors.variables.deleteInput).type('delete')
    cy.get(selectors.variables.confirmDeleteButton).click()
    cy.get(selectors.variables.successMessage).should('have.text', 'Variable successfully deleted')
```

File: cypress/e2e/edge_application.cy.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
+const selectors = {
```
Comment (alisson-marcelo-azion) [inline cypress/e2e/edge_application.cy.js:1]: Estes seletores poderiam ser colocados em um arquivo separado.

**Review by william-tome — CHANGES_REQUESTED** [review]
Ainda tem alguns `;`

**Review by alisson-marcelo-azion — APPROVED** [review]

**Review by alisson-marcelo-azion — CHANGES_REQUESTED** [review]
Testes E2E Falhando

File: cypress/e2e/edge_application.cy.js (hunk 5 above)
Comment (alisson-marcelo-azion) [reply cypress/e2e/edge_application.cy.js:1]: Pode ser outro

**Comment by alisson-marcelo-azion** [comment]
Restaura de forma completa o cypress/e2e/variables.cy.js. Agora ele está sendo gerado dinamicamente 😉

File: cypress/e2e/edge_application.cy.js
Comment (alisson-marcelo-azion) [inline cypress/e2e/edge_application.cy.js:1]: Acho que esse arquivo deveria ter hífen no nome _cypress/e2e/edge-application.cy.js_

**Pull request merged**